dev-client/src/screens/SoilScreen/ColorScreen/components/ManualWokrflow.tsx

```diff
@@ -0,0 +1,257 @@
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {SoilPitInputScreenProps} from 'terraso-mobile-client/screens/SoilScreen/components/SoilPitInputScreenScaffold';
+import {
+  Box,
+  Column,
+  Heading,
+  Paragraph,
+  Row,
+} from 'terraso-mobile-client/components/NativeBaseAdapters';
+import {useTranslation} from 'react-i18next';
+import {pitMethodSummary} from 'terraso-mobile-client/screens/SoilScreen/utils/renderValues';
+import {useDispatch, useSelector} from 'terraso-mobile-client/store';
+import {selectDepthDependentData} from 'terraso-client-shared/selectors';
+import {SwitchWorkflowButton} from 'terraso-mobile-client/screens/SoilScreen/ColorScreen/components/SwitchWorkflowButton';
+import {DepthDependentSelect} from 'terraso-mobile-client/screens/SoilScreen/components/DepthDependentSelect';
+import {
+  updateDepthDependentSoilData,
+  SoilColorHue,
+  ColorHueSubstep,
+  soilColorHues,
+  ColorValue,
+  ColorChroma,
+} from 'terraso-client-shared/soilId/soilIdSlice';
+import {Icon} from 'terraso-mobile-client/components/Icons';
+import {useCallback, useMemo, useState} from 'react';
+import {Select} from 'terraso-mobile-client/components/inputs/Select';
+import {
+  parseMunsellHue,
+  renderMunsellHue,
+} from 'terraso-mobile-client/screens/SoilScreen/ColorScreen/utils/munsellConversions';
+import Animated, {LinearTransition} from 'react-native-reanimated';
+import {StyleSheet} from 'react-native';
+import {
+  isValidChromaFor,
+  isValidSubstepFor,
+  isValidValueFor,
+  validComponents,
+} from 'terraso-mobile-client/screens/SoilScreen/ColorScreen/utils/soilColors';
+
+export const ManualWorkflow = (props: SoilPitInputScreenProps) => {
+  const {t} = useTranslation();
+  const dispatch = useDispatch();
+  const data = useSelector(selectDepthDependentData(props));
+  const {complete} = pitMethodSummary(t, data, 'soilColor');
+
+  const {substep: initialSubstep, hue: initialHue} = useMemo(
+    () =>
+      typeof data?.colorHue === 'number'
+        ? renderMunsellHue({
+            colorHue: data?.colorHue,
+            colorChroma: data?.colorChroma,
+          })
+        : {substep: null, hue: null},
+    [data],
+  );
+
+  const [selectedSubstep, setSelectedSubstep] = useState(
+    initialSubstep ?? null,
+  );
+  const [selectedHue, setSelectedHue] = useState(
+    (initialHue ?? null) as SoilColorHue | null,
+  );
+  const [selectedValue, setSelectedValue] = useState(
+    (data?.colorValue ?? null) as ColorValue | null,
+  );
+
+  const updateColor = useCallback(
+    (
+      color:
+        | {hue: SoilColorHue | null}
+        | {substep: ColorHueSubstep | null}
+        | {value: ColorValue | null},
+    ) => {
+      let hue = selectedHue;
+      let substep = selectedSubstep;
+      let value = selectedValue;
+      let chroma = (data?.colorChroma ?? null) as ColorChroma | 0 | null;
+      if ('value' in color) {
```

Comment:
comment: I feel that lines 93 - 121 (and their associated helper methods) really should be their own function -- they don't appear to be React specific and seem like something that should be tested in isolation at some point. (I also would request a commend explaining what this logic is doing; it doesn't seem obvious to me what the operation is here.(